### PR TITLE
Pass UrlFile object to Provider.checkFileUrl instead of string

### PR DESCRIPTION
**Pull request by MantisClone**
Changes proposed in this PR:

- Pass `UrlFile` object to `Provider.checkFileUrl` instead of a string

Depends on https://github.com/oceanprotocol/ocean.js/pull/1627

Reasoning:

This is an incremental step towards supporting other file storage types like Arweave, Filecoin, etc...

### CODE REVIEW

**Comment by MantisClone** [comment]
Hello, @kremalicious Thanks for taking a look at this and approving the CI checks to run.  I believe the CI checks are failing because this PR depends on https://github.com/oceanprotocol/ocean.js/pull/1627 which has not yet been merged/released.

**Comment by MantisClone** [comment]
Hello @alexcos20 @kremalicious @mihaisc @bogdanfazakas

This PR depends on https://github.com/oceanprotocol/ocean.js/pull/1627, but other than that, I think it is ready for review.

File: src/@utils/provider.ts [hunk 1]
```diff
@@ -83,11 +84,11 @@ export async function getFileDidInfo(
 }
 
 export async function getFileUrlInfo(
```
Comment (kremalicious) [inline src/@utils/provider.ts:86]: Same as over on ocean.js, if it does more than checking URLs, method needs a new name. Here we can simply remove `URL` and end up with `getFileInfo()`

File: src/components/@shared/FormFields/FilesInput/index.tsx [hunk 2]
```diff
@@ -23,7 +23,12 @@ export default function FilesInput(props: InputProps): ReactElement {
         ? values?.services[0].providerUrl.url
         : asset.services[0].serviceEndpoint
       setIsLoading(true)
-      const checkedFile = await getFileUrlInfo(url, providerUrl)
+      const urlFile: UrlFile = {
+        type: 'url',
+        url: url,
```
Comment (kremalicious) [inline src/components/@shared/FormFields/FilesInput/index.tsx:28]: shorthand notation please, which then most likely allows us to have this short line in one line, like:

```js
const urlFile: UrlFile = { type: 'url', url, method: 'GET' }
```

**Comment by kremalicious** [comment]
Misses one more usage of that file check:
https://github.com/oceanprotocol/market/blob/83d83447b6f241740d3fdd1c3b78f99a01b3f5c9/src/components/Asset/AssetActions/index.tsx#L44-L82

**Comment by kremalicious** [comment]
ocean.js v2.3.0 has the changes required for this PR

File: src/@utils/provider.ts (hunk 1 above)
Comment (MantisClone) [reply src/@utils/provider.ts]: Done.

**Comment by MantisClone** [comment]
> Misses one more usage of that file check, be sure to handle this for publish preview & published asset which is what the condition for `fileInfoResponse` in here is for:
> 
> https://github.com/oceanprotocol/market/blob/83d83447b6f241740d3fdd1c3b78f99a01b3f5c9/src/components/Asset/AssetActions/index.tsx#L44-L82

I looked at this briefly, but I don't know how to fix it.

File: package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "@coingecko/cryptoformat": "^0.5.4",
     "@loadable/component": "^5.15.2",
     "@oceanprotocol/art": "^3.2.0",
-    "@oceanprotocol/lib": "^2.2.3",
+    "@oceanprotocol/lib": "^2.3.0",
```
Comment (kremalicious) [inline package.json:29]: need to run `npm install` so `package-lock.json` updates, then commit that file too for actual upgrade

**Comment by kremalicious** [comment]
This needs merge or rebase against `main` and then continue. No need to update ocean.js in here then.

> I looked at this briefly, but I don't know how to fix it.

ok, what did you try to solve this? What exactly is unclear?

**Comment by kremalicious** [comment]
closing as abandoned, and #1765 is picking things up from here.

**Pull request closed without merging**